src/daft-micropartition/src/micropartition.rs

```diff
@@ -31,6 +31,14 @@ pub(crate) enum TableState {
     Loaded(Arc<Vec<Table>>),
 }
 
+pub fn char_to_byte(char_val: Option<char>) -> Result<Option<u8>, Error> {
+
+    match u8::try_from(char_val.unwrap()){
+        Err(_e) => Err(Error::WrongChar{val: char_val.unwrap()}),
+        Ok(char_val) => Ok(Some(char_val)),
+    }
```

Comment:
`char_val.unwrap()` will cause this to panic of `char_val` is `None`, which may very well happen for `delimiter`, `quote`, `escape_char`, or `comment`! Instead of unwrapping, we should short-circuit by returning `Ok(None)` if `char_val` is `None`. A nice pattern for this would be using `Option::map(u8::try_from)` to turn `char_val` into an `Option<Result<u8, TryFromCharError>>`, then `.transpose()` to turn this into a `Result<Option<u8>, TryFromCharError>`, and then `.context(WrongCharSnafu)` to turn this into a `Result<Option<u8>, super::Error>`.
```suggestion
    char_val.map(u8::try_from).transpose().context(WrongCharSnafu)
```